build/README.md

```diff
@@ -0,0 +1,20 @@
+## Images
+
+### How the images are built
+
+`cert-manager` images in `quay.io` are multi-arch images for a number of `linux` architectures.
+The individual container bundles for each architecture are built using `Bazel` functionality in this repository.
+Docker [manifest list](https://docs.docker.com/registry/spec/manifest-v2-2/#manifest-list) is then created by [`cmrel`](https://github.com/cert-manager/release) and the arch-specific container bundles and the manifest list pushed to `quay.io`.
+Therefore the `multi_arch..`-named rules in this repository don't refer to 'multi-arch' in a sense of creating multi-arch images (i.e manifest lists) and the functionality related to pushing images is mostly unused.
+
```

Comment:
Give some examples and use-cases for building images with cgo enabled / disabled and examples use-cases for "dynamic base image"
Also explain what a dynamic base image is.

Reply:
I think I'd echo @wallrj's comment here - it'd be super useful to have a little more info here for users who're investigating the build process, and maybe an example of how to build with a static / dynamic base image.

Maybe something like:

```suggestion
### Base image types

By default, cert-manager binaries are built statically linked, with cgo disabled. Binaries are packaged in a [static distroless base image](https://github.com/GoogleContainerTools/distroless). Unless you're making changes to cert-manager itself, this default is probably exactly what you want.

In some scenarios - such as if you need to link against a different TLS library - you might want to enable cgo and use a dynamic distroless base image. To do that, <INSERT INSTRUCTIONS HERE>.
```